src/test/regress/expected/background_rebalance.out

```diff
@@ -0,0 +1,180 @@
+ citus_rebalance_start
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_rebalance_wait();
+ citus_rebalance_wait
+---------------------------------------------------------------------
+
+(1 row)
+
+DROP TABLE t2;
+SELECT citus_move_shard_placement(85674000, 'localhost', :worker_1_port, 'localhost', :worker_2_port, shard_transfer_mode => 'block_writes');
+ citus_move_shard_placement
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show we can stop a rebalance, the stop causes the move to not have happened, eg, our move back below fails.
+SELECT citus_rebalance_start();
+NOTICE:  Scheduled 1 moves as job xxx
+DETAIL:  Rebalance scheduled as background job
+HINT:  To monitor progress, run: SELECT * FROM pg_dist_background_task WHERE job_id = xxx ORDER BY task_id ASC; or SELECT * FROM get_rebalance_progress();
+ citus_rebalance_start
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_rebalance_stop();
+ citus_rebalance_stop
+---------------------------------------------------------------------
+
+(1 row)
+
+-- waiting on this rebalance is racy, as it sometimes sees no rebalance is ongoing while other times it actually sees it ongoing
+-- we simply sleep a bit here
+SELECT pg_sleep(1);
+ pg_sleep
+---------------------------------------------------------------------
+
+(1 row)
+
+-- failing move due to a stopped rebalance, first clean orphans to make the error stable
+SET client_min_messages TO WARNING;
+CALL citus_cleanup_orphaned_shards();
+RESET client_min_messages;
+SELECT citus_move_shard_placement(85674000, 'localhost', :worker_1_port, 'localhost', :worker_2_port, shard_transfer_mode => 'block_writes');
+WARNING:  shard is already present on node localhost:xxxxx
+DETAIL:  Move may have already completed.
+ citus_move_shard_placement
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show we can't start the rebalancer twice
+SELECT citus_rebalance_start();
+NOTICE:  Scheduled 1 moves as job xxx
+DETAIL:  Rebalance scheduled as background job
+HINT:  To monitor progress, run: SELECT * FROM pg_dist_background_task WHERE job_id = xxx ORDER BY task_id ASC; or SELECT * FROM get_rebalance_progress();
+ citus_rebalance_start
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_rebalance_start();
+ERROR:  A rebalance is already running as job xxx
+DETAIL:  A rebalance was already scheduled as background job
+HINT:  To monitor progress, run: SELECT * FROM pg_dist_background_task WHERE job_id = xxx ORDER BY task_id ASC; or SELECT * FROM get_rebalance_progress();
+SELECT citus_rebalance_wait();
+ citus_rebalance_wait
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show that the old rebalancer cannot be started with a background rebalance in progress
```

Comment:
It would be good to also test the opposite in an isolation test.

Reply:
Let me see if I can find a test to add it to. Setting up a complete isolation test for this is quite cumbersome

Reply:
added to an existing isolation test